### PR TITLE
Added Profiler View and Removed update parts name on drop elements

### PR DESCRIPTION
**Pull request by igorkorsukov**
For view Profile data - go to Diagnostic->System->Show profiler...

image

About removing update parts name on drop elements
There are several problems:
* The parts names were updated every time any element was dropped from the Palette to Score
* Each time the names of all parts are updated ( only need for which we added an instrument change)
* Updating the names of parts, if the new name differs from the current one, is carried out slowly, respectively, when updating the names of all parts, this can take more than 30 seconds
* It is not correct to update the name of the part for a drop element from the Palette, because we can also edit an already added "instrument change". Accordingly, in this case, we should also update the name of the part. Accordingly, part name updates when adding a "instrument change" and editing a "instrument change" should be consistent.

In general, the зart name update needs to be re-implemented properly. @DmitryArefiev FYI

### CODE REVIEW

File: src/diagnostics/view/system/profilerviewmodel.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        item.group = group;
+        item.data = data;
+
+        m_allList.append(item);
+    }
+
+    find(m_searchText);
+}
+
+void ProfilerViewModel::find(const QString& str)
```
Comment (RomanPudashkin) [inline src/diagnostics/view/system/profilerviewmodel.cpp:94]: The name of this function is confusing. I would suggest renaming it to load (or something similar). You can also use SortFilterProxyModel in such situations

File: src/diagnostics/view/system/profilerviewmodel.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    QStringList list = str.split("\n");
+    foreach (const QString& data, list) {
+        Item item;
+        item.group = group;
+        item.data = data;
+
+        m_allList.append(item);
+    }
+
+    group = "Other thread";
```
Comment (RomanPudashkin) [inline src/diagnostics/view/system/profilerviewmodel.cpp:80]: This block of code duplicates the block above. I think you can add a lambda with this code and reuse it twice

**Review by RomanPudashkin — APPROVED** [review]

File: src/diagnostics/view/system/profilerviewmodel.cpp (hunk 1 above)
Comment (igorkorsukov) [reply src/diagnostics/view/system/profilerviewmodel.cpp:94]: This method is find (filtering), not load data (data already loaded to `m_allList`)

I don't like QSortFilterProxyModel, if I can access and can edit model source.

**Comment by DmitryArefiev** [comment]
Tested #9094 8440 on master build (Win10, Mac11, LinuxMint 20) - FIXED

**Pull request merged**